### ISSUE TITLE
Kubernetes python API calls and reponse processing

### ISSUE THREAD

**Issue by cmonfort** [scality/metalk8s#1047]
In many places in `metalk8s/salt/_modules/metalk8s_kubernetes.py`, calls and response processing are sub-optimal, for example:
```
def nodes(**kwargs):
    cfg = _setup_conn(**kwargs)
    try:
        api_instance = kubernetes.client.CoreV1Api()
        api_response = api_instance.list_node()

        return [k8s_node['metadata']['name'] for k8s_node in api_response.to_dict().get('items')]
```
there is no need to convert to dict(), and the returned value should be:
`return [k8s_node.metadata.name for k8s_node in api_response.items]`
because even when nothing is found a valid reponse - with an empty `response.items` array is returned, and no error is fired.

In a similar way:
```
def node(name, **kwargs):
    cfg = _setup_conn(**kwargs)
    try:
        api_instance = kubernetes.client.CoreV1Api()
        api_response = api_instance.list_node()
    except (ApiException, HTTPError) as exc:
        if isinstance(exc, ApiException) and exc.status == 404:
            return None
        else:
            log.exception('Exception when calling CoreV1Api->list_node')
            raise CommandExecutionError(exc)
    finally:
        _cleanup(**cfg)

    for k8s_node in api_response.items:
        if k8s_node.metadata.name == name:
            return k8s_node.to_dict()

    return None
```
We loop on all the returned nodes to find out the one we want, whereas we could have use `field_selector` and let Kubernetes do the job:
`api_response = api_instance.list_node(field_selector='metadata.name={0}'.format(name))`
then the `api_response.items` contains only the targeted node, or nothing... (extra question: do we need/is it better to convert response to dict(), or directly use the node object returned ?)

Note: those are just examples, and the same patterns are used in many different places in the file.

**Comment by gdemonet** [comment]
Those directly come from the Saltstack `kubernetes` execution module, which we only forked and replicated for the additional features that were missing.
This whole module needs refactoring, not only about optimizing performance of calls, but also to ease maintenance and debugging. Let's create a ticket for tracking the refactoring of this module, and reference this one in it.

However, I disagree that this is debt (since not introduced by us) - it's an enhancement.
I'm removing the milestone as well, since we need to perform triage among debt/enhancement tickets before deciding what can be included in our milestones.

**Comment by TeddyAndrieux** [comment]
Fixed by #1972
